rllib/evaluation/sampler.py

```diff
@@ -967,10 +967,12 @@ def _process_observations(
             # Horizon hit and we have a soft horizon (no hard env reset).
             if hit_horizon and soft_horizon:
                 episode.soft_reset()
-                resetted_obs: Dict[AgentID, EnvObsType] = all_agents_obs
+                resetted_obs: Dict[EnvID, Dict[AgentID, EnvObsType]] = \
+                    {env_id: all_agents_obs}
             else:
                 del active_episodes[env_id]
-                resetted_obs: Dict[AgentID, EnvObsType] = base_env.try_reset(
+                resetted_obs: Dict[EnvID, Dict[AgentID, EnvObsType]] = \
```

Comment:
I changed the code for handling obs from soft resets to match the return type of the base_env.try_reset() function